pkg/cli/cmd/run/run.go

```diff
@@ -215,3 +258,33 @@ func (r *Runner) displayPortforwardMessages(status <-chan portforward.StatusMess
 		fmt.Printf("%s %s [port-forward] %s from localhost:%d -> ::%d\n", regular.Sprint(message.ReplicaName), bold.Sprint(message.ContainerName), message.Kind, message.LocalPort, message.RemotePort)
 	}
 }
+
+// dashboardDeploymentExists checks if a dashboard deployment exists in the given Kubernetes context.
+func dashboardDeploymentExists(ctx context.Context, kubernetesClient k8sclient.Interface, kubeContext string, dashboardLabelSelector labels.Selector) bool {
+	var client k8sclient.Interface
+	var err error
+
+	// If a kubernetes client is provided, use it. Otherwise, create a new client.
+	// This helps with testing.
+	if kubernetesClient != nil {
+		client = kubernetesClient
+	} else {
+		client, _, err = kubernetes.NewClientset(kubeContext)
+		if err != nil {
+			return false
+		}
+	}
```

Comment:
No need to declare client variable here. 

```suggestion
	if kubernetesClient == nil {
		kubernetesClient, _, err = kubernetes.NewClientset(kubeContext)
		if err != nil {
			return false
		}
	}
```